src/corecel/sys/TracingSession.cc

```diff
@@ -0,0 +1,154 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/sys/TracingSession.cc
+//! \brief RAII class for managing a perfetto session and its resources.
+//---------------------------------------------------------------------------//
+#include "TracingSession.hh"
+
+#include <fcntl.h>
+#include <perfetto.h>
+
+#include "corecel/Assert.hh"
+
+#include "Environment.hh"
+#include "ScopedProfiling.hh"
+
+#include "detail/TrackEvent.perfetto.hh"
+
+PERFETTO_TRACK_EVENT_STATIC_STORAGE();
+
+namespace
+{
+//---------------------------------------------------------------------------//
+using celeritas::TracingMode;
```

Comment:
Is there a reason not to just put this anonymous namespace inside the celeritas one like we usually do?